core/src/main/kotlin/io/specmatic/core/pattern/XMLPattern.kt

```diff
@@ -103,6 +103,11 @@ data class XMLPattern(override val pattern: XMLTypeData = XMLTypeData(realName =
         xmlValue.name.substringAfter(":") == this.pattern.name
 
     override fun matches(sampleData: Value?, resolver: Resolver): Result {
+        // Special handling for empty string when expecting XML (common in test scenarios)
```

Comment:
Not the right fix, roll this back.

I suspect something is wrong with the `generate` or `newBasedOn` methods.